File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/MenuTitleItemFactory.java
package me.xiaopan.sketchsample.adapter.itemfactory;

import android.content.Context;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory;
import me.xiaopan.sketchsample.R;
import me.xiaopan.sketchsample.adapter.BindAssemblyRecyclerItem;

public class MenuTitleItemFactory extends AssemblyRecyclerItemFactory<MenuTitleItemFactory.MenuTitleItem> {
    @Override
    public boolean isTarget(Object o) {
        return o instanceof String;
    }

    @Override
    public MenuTitleItem createAssemblyItem(ViewGroup viewGroup) {
        return new MenuTitleItem(R.layout.list_item_menu_title, viewGroup);
    }

    public class MenuTitleItem extends BindAssemblyRecyclerItem<String> {
        @BindView(R.id.text_menuTitleItem_title)
        TextView textView;

        public MenuTitleItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onConfigViews(Context context) {

        }

        @Override
        protected void onSetData(int i, String title) {
            textView.setText(title);
        }
    }
}

File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/MenuTitleItemFactory.kt
package me.xiaopan.sketchsample.adapter.itemfactory

import android.content.Context
import android.view.ViewGroup
import android.widget.TextView
import me.xiaopan.assemblyadapter.AssemblyRecyclerItem
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory
import me.xiaopan.sketchsample.R
import me.xiaopan.ssvt.bindView

class MenuTitleItemFactory : AssemblyRecyclerItemFactory<MenuTitleItemFactory.MenuTitleItem>() {
    override fun isTarget(o: Any): Boolean {
        return o is String
    }

    override fun createAssemblyItem(viewGroup: ViewGroup): MenuTitleItem {
        return MenuTitleItem(R.layout.list_item_menu_title, viewGroup)
    }

    inner class MenuTitleItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyRecyclerItem<String>(itemLayoutId, parent) {
        val textView: TextView by bindView(R.id.text_menuTitleItem_title)

        override fun onConfigViews(context: Context) {

        }

        override fun onSetData(i: Int, title: String) {
            textView.text = title
        }
    }
}
